chapters/02-current-state.Rmd

```diff
@@ -1,7 +1,27 @@
 # Current state of the art {#currentState}
 
-...
+There are several options available for computing univariate kernel density estimates in Python.
 
-For humongous data streams (like at CERN @cern), Kernel Density Estimation itself needs to be approximated.
+The most popular ones are SciPy's `gaussian_kde`[@ScipyStatsGaussian], Statsmodels' `KDEUnivariate`[@KernelDensityEstimation] as well as `KernelDensity` from the Scikit-learn package[@SklearnNeighborsKernelDensity].
 
-...
\ No newline at end of file
+As Jake VanderPlas was able to show in his comparison[@vanderplas2013] Scikit-learn's tree based approach to compute the kernel density estimation was the most efficient in the vast majority of cases in 2013.
+The question of the optimal KDE implementation for any situation, however, is not entirely straightforward, and depends a lot on what your particular goals are.
+
+Statsmodels includes a computation based on Fast Fourier Transform (FFT) and normal reference rules for choosing the optimal bandwidth, which Scikit-learns package lacks for instance.
+
+In 2018 Tommy Odland proposed an entirely new implementation called KDEpy[@KDEpy] which incorporates features of all implementations mentioned before as well as additional kernels and an additional method to calculate the bandwidth using the Improved Sheather Jones (ISJ) algorithm first proposed by Botev et al[@botev2010kernel].
+
+He was also able to show that his FFT based computation was able to outperform previous implementations (even Scikit-learn's tree based approach) in terms of runtime by a factor of at least one order of magnitude, making KDEpy the de-facto standard of Kernel Density Estimation in Python.
```

Comment:
[7] That's nice, but I miss somewhat the flow overall. The previous paragraphs reed somewhat more like single statements, en "rote fade" is somewhat missing I find